src/duke/main/Duke.java

```diff
@@ -0,0 +1,223 @@
+package duke.main;
+
+import duke.task.Deadline;
+import duke.task.Task;
+import duke.task.Event;
+import duke.task.Todo;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Duke{
+    public static final String LINEBREAK = "____________________________________________________________";
+    public static final String FILEPATH = "data/duke.txt";
+    public static final String DIRPATH = "data";
+    public static ArrayList<Task> list = new ArrayList<>();
+    public static Scanner in = new Scanner(System.in);
+    public static int taskCount = 0;
+    public static boolean isLoaded = false;
+    public static void loadSavedTasks(){
+        try{
+            File dirPath = new File(DIRPATH);
+            if (!dirPath.exists()){
+                dirPath.mkdirs();
+            }
+            File file = new File(FILEPATH);
+            Scanner txtScanner = new Scanner(file);
+            while (txtScanner.hasNext()) {
+                isLoaded = true;
+                String line = txtScanner.nextLine();
+                String[] fields = line.split(" \\| ");
+                switch(fields[0]){
+                case "T":
+                    list.add(new Todo(fields[2], fields[1].equals("1")));
+                    break;
+                case "D":
+                    list.add(new Deadline(fields[2], fields[1].equals("1"), fields[3]));
+                    break;
+                case "E":
+                    list.add(new Event(fields[2], fields[1].equals("1"), fields[3]));
+                    break;
+                default:
+                    throw new DukeException("Error: Invalid file format.");
+                }
+                taskCount++;
+            }
+        }catch (IOException e){
+            System.out.println(LINEBREAK);
+            System.out.println(" Error: " + e.getMessage());
+            System.out.println(LINEBREAK);
+            return;
+        }catch (DukeException e){
+            System.out.println(LINEBREAK);
+            System.out.println(e.getMessage());
+            System.out.println(LINEBREAK);
+            return;
+        }
```

Comment:
[Don't Repeat Yourself](https://nus-cs2113-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication). You might want to consider extracting (or combining) them, as they have almost the same functionality.